cmd/promtool/main.go

```diff
@@ -66,15 +65,23 @@ func main() {
 
 	queryCmd := app.Command("query", "Run query against a Prometheus server.")
 	queryInstantCmd := queryCmd.Command("instant", "Run instant query.")
-	queryServer := queryInstantCmd.Arg("server", "Prometheus server to query.").Required().URL()
+	queryServer := queryInstantCmd.Arg("server", "Prometheus server to query.").Required().String()
 	queryExpr := queryInstantCmd.Arg("expr", "PromQL query expression.").Required().String()
 
 	queryRangeCmd := queryCmd.Command("range", "Run range query.")
-	queryRangeServer := queryRangeCmd.Arg("server", "Prometheus server to query.").Required().URL()
+	queryRangeServer := queryRangeCmd.Arg("server", "Prometheus server to query.").Required().String()
```

Comment:
Maybe this has been discussed in one of the many outdated comment threads, but what was the reason here for using `String()` instead of `URL()`?

Reply:
Since I didn't make any change there so you can still see above. 
>The [constructor](https://github.com/prometheus/client_golang/blob/master/api/client.go#L66) takes the [argument](https://github.com/prometheus/client_golang/blob/master/api/client.go#L43) of string type and it returns error if it fails to parse the URL string. So I think it's redundant that it parses the url string by `kingpin.URL()` method in advance.

Reply:
Ah ok, hm yeah. Although IMO it's still nice to be as specific as possible in flag definitions, even if it might implicitly lead to checking the URL twice somewhere later in the program.